service/raft/raft_group_registry.hh

```diff
@@ -49,6 +71,7 @@ struct raft_server_for_group {
     raft_rpc& rpc;
     raft_sys_table_storage& persistence;
     std::optional<seastar::future<>> aborted;
+    std::optional<lowres_clock::duration> timeout;
```

Comment:
I'm a bit confused, there's 2 classes now that store a timeout and then there's the timeout parameter passed to API functions.

I wonder if it wouldn't be simpler to have `raft_server_with_timeouts` **own** `raft::server`, and then `raft_server_for_group` would hold a `unique_ptr` to `raft_server_with_timoeuts`.

We could implement `raft_server_with_timeout` through private inheritance from `raft::server`. And then it would selectively expose the interface of `raft::server` (functions like `stepdown`) vs its own (the timeout variants of certain functions like `add_entry`).

Just something to consider.

Reply:
> Why inherit privately from raft::server?

Does the hierarchy satisfy LSP? Maybe it does (didn't think too much about it), in which case public would be OK too.